File: program/program/convert-octal-to-binary/convertOctalToBinary.scala
object NumberConv {
def binaryCon(n : Int) =
{
var i=0
var num=n
var A=new Array[Int](10)
while(num>0)
{
A(i)=num%2
i=i+1
num=num/2
}
println("Binary equivalent :" )
for(j<-i-1 to 0 by -1)
print(A(j))
}

def main(args: Array[String]) = 
{
println("Enter a Octal number :")
var n=scala.io.StdIn.readInt()
println("%d".format(n))
binaryCon(n)
}
}

File: program/program/convert-octal-to-binary/convert_octal_to_binary.hs
import Data.Char (digitToInt)

-- Convert an octal digit to its binary representation
octalToBinaryDigit :: Char -> String
octalToBinaryDigit '0' = "000"
octalToBinaryDigit '1' = "001"
octalToBinaryDigit '2' = "010"
octalToBinaryDigit '3' = "011"
octalToBinaryDigit '4' = "100"
octalToBinaryDigit '5' = "101"
octalToBinaryDigit '6' = "110"
octalToBinaryDigit '7' = "111"
octalToBinaryDigit _   = error "Invalid octal digit"

-- Convert an octal number string to binary
octalToBinary :: String -> String
octalToBinary = concatMap octalToBinaryDigit

main :: IO ()
main = do
    putStrLn "Enter an octal number:"
    octal <- getLine
    let binary = octalToBinary octal
    putStrLn $ "Binary equivalent: " ++ binary
